CardForm/src/main/java/com/braintreepayments/api/CardType.java

```diff
@@ -11,228 +11,16 @@
-            return relaxedPrefixPatternMatch;
-        }
-
-        if (!cardNumber.isEmpty()) {
-            return UNKNOWN;
-        }
-
-        return EMPTY;
-    }
-
-    private static CardType forCardNumberPattern(String cardNumber) {
-        for (CardType cardType : values()) {
-            if (cardType.getPattern().matcher(cardNumber).matches()) {
-                return cardType;
-            }
-        }
-
-        return EMPTY;
-    }
-
-    private static CardType forCardNumberRelaxedPrefixPattern(String cardNumber) {
-        for (CardType cardTypeRelaxed : values()) {
-            if (cardTypeRelaxed.getRelaxedPrefixPattern() != null) {
-                if (cardTypeRelaxed.getRelaxedPrefixPattern().matcher(cardNumber).matches()) {
-                    return cardTypeRelaxed;
-                }
-            }
-        }
-
-        return EMPTY;
-    }
-
-    /**
-     * @return The regex matching this card type.
-     */
-    public Pattern getPattern() {
-        return mPattern;
-    }
-
-    /**
-     * @return The relaxed prefix regex matching this card type. To be used in determining card type if no pattern matches.
-     */
-    public Pattern getRelaxedPrefixPattern() {
-        return mRelaxedPrefixPattern;
-    }
-
-    /**
-     * @return The android resource id for the front card image, highlighting card number format.
-     */
-    public int getFrontResource() {
-        return mFrontResource;
-    }
-
-    /**
-     * @return The android resource id for the security code name for this card type.
-     */
-    public int getSecurityCodeName() {
-        return mSecurityCodeName;
-    }
-
-    /**
-     * @return The length of the current card's security code.
-     */
-    public int getSecurityCodeLength() {
-        return mSecurityCodeLength;
-    }
-
-    /**
-     * @return minimum length of a card for this {@link CardType}
-     */
-    public int getMinCardLength() {
-        return mMinCardLength;
-    }
-
-    /**
-     * @return max length of a card for this {@link CardType}
-     */
-    public int getMaxCardLength() {
-        return mMaxCardLength;
-    }
-
-    /**
-     * @return the locations where spaces should be inserted when formatting the card in a user
-     * friendly way. Only for display purposes.
-     */
-    public int[] getSpaceIndices() {
-        return this == AMEX ? AMEX_SPACE_INDICES : DEFAULT_SPACE_INDICES;
-    }
-
-    /**
-     * Performs the Luhn check on the given card number.
-     *
-     * @param cardNumber a String consisting of numeric digits (only).
-     * @return {@code true} if the sequence passes the checksum
-     * @throws IllegalArgumentException if {@code cardNumber} contained a non-digit (where {@link
-     * Character#isDefined(char)} is {@code false}).
-     * @see <a href="http://en.wikipedia.org/wiki/Luhn_algorithm">Luhn Algorithm (Wikipedia)</a>
-     */
-    public static boolean isLuhnValid(String cardNumber) {
```

Comment:
Since this is all going into a `next-major-version` branch is it worth explicitly calling out these breaking changes in the changelog? On iOS we have a [breaking changes](https://github.com/braintree/braintree_ios/blob/master/CHANGELOG.md#unreleased) section where we've listed items like this. I realize it's a lot of functions to call out but since there isn't a docstring calling out it could be removed at any time, it may be helpful. Thoughts?

Reply:
This is a good callout. I'll add some CHANGELOG notes that it's no longer in the public facing API.